sections/featured-product.liquid

```diff
@@ -0,0 +1,591 @@
+{{ 'section-featured-product.css' | asset_url | stylesheet_tag }}
+{{ 'component-accordion.css' | asset_url | stylesheet_tag }}
+{{ 'component-price.css' | asset_url | stylesheet_tag }}
+{{ 'component-rte.css' | asset_url | stylesheet_tag }}
+
+<link rel="stylesheet" href="{{ 'component-deferred-media.css' | asset_url }}" media="print" onload="this.media='all'">
+
+{%- assign product = section.settings.product -%}
+
+{%- assign first_3d_model = product.media | where: "media_type", "model" | first -%}
+{%- if first_3d_model -%}
+  {{ 'component-product-model.css' | asset_url | stylesheet_tag }}
+  <link id="ModelViewerStyle" rel="stylesheet" href="https://cdn.shopify.com/shopifycloud/model-viewer-ui/assets/v1.0/model-viewer-ui.css" media="print" onload="this.media='all'">
+  <link id="ModelViewerOverride" rel="stylesheet" href="{{ 'component-model-viewer-ui.css' | asset_url }}" media="print" onload="this.media='all'">
+{%- endif -%}
+
+<section class="{% if section.settings.secondary_background %}background-secondary{% else %}spaced-section{% endif %}">
+  <div class="page-width">
+    <div class="featured-product product grid grid--1-col {% if product.media.size > 0 %}grid--2-col-tablet{% else %}product--no-media{% endif %}">
+      <div class="grid__item product__media-wrapper">
+        <a class="skip-to-content-link button visually-hidden" href="#ProductInfo-{{ section.id }}">
+          {{ "accessibility.skip_to_product_info" | t }}
+        </a>
+        <ul class="product__media-list grid grid--peek list-unstyled" role="list">
+          {%- if product.selected_or_first_available_variant.featured_media != null -%}
+            {%- assign media = product.selected_or_first_available_variant.featured_media -%}
+            <li class="product__media-item grid__item{% if media.media_type != 'image' %} product__media-item--full{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
+              {% render 'product-thumbnail', media: media, position: 'featured', loop: section.settings.enable_video_looping, modal_id: section.id %}
+            </li>
+          {%- endif -%}
+          {%- liquid
+            assign media_to_render = product.featured_media.id
+            for variant in product.variants
+              assign media_to_render = media_to_render | append: variant.featured_media.id | append: ' '
+            endfor
+          -%}
+          {%- for media in product.media -%}
+            {%- if media_to_render contains media.id and media.id != product.selected_or_first_available_variant.featured_media.id -%}
+              <li class="product__media-item grid__item{% if media.media_type != 'image' %} product__media-item--full{% endif %}" data-media-id="{{ section.id }}-{{ media.id }}">
+                {% render 'product-thumbnail', media: media, position: forloop.index, loop: section.settings.enable_video_looping, modal_id: section.id %}
+              </li>
+            {%- endif -%}
+          {%- endfor -%}
```

Comment:
I'm curious to know why we need both `li` logics here 🤔  I feel like I'm missing something obvious.

Reply:
We're using a CSS strategy based on `:first-child` here to decide which media to display, as the variant-select scripts will always move the current selected media to the top of the list. There's no toggling of `hidden` and `visible` classes, as that isn't needed on the product page.


So, for the featured product: the first `<li>` ensures that **the first media** will always be the one attached to the currently selected variant (`selected_or_first_available_variant`). 

Consider a product with:
- Two variants. 
- Both of them have a featured media
- The first variant is sold out, so the second is the first available variant.
 
Because of that, the second variant will be selected by default. However, if we use `for media in product.media`, the first media it'll render won't be the one attached to the current variant, if that makes sense. 👀 

If you think adding a `hidden`/`visible` class logic is better, let me know. I preferred this way to avoid any modifications to the shared variant JS script.